hoomd/md/pytest/test_half_step_hook.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Test that `HalfStepHook` works."""
+
+import numpy as np
+import pytest
+import hoomd
+
+from hoomd import md
+
+
+class DistanceCV(md.HalfStepHook):
+
+    def __init__(self, sim):
+        super().__init__()
```

Comment:
Try this:
```suggestion
        md.HalfStepHook.__init__()
```
This is the pattern we use for `Variant` and `Trigger` with success. 

From the pybind11 docs: https://pybind11.readthedocs.io/en/stable/advanced/classes.html
> Note that a direct __init__ constructor should be called, and super() should not be used. For simple cases of linear inheritance, super() may work, but once you begin mixing Python and C++ multiple inheritance, things will fall apart due to differences between Python’s MRO and C++’s mechanisms.

Reply:
Oh. I had missed that part of the docs (and we haven't need this before). Fixed now.

Reply:
I hope that fixes your build errors.

Reply:
Sorry, I got that suggestion wrong. It should be `md.HalfStepHook.__init__(self)`

Reply:
Fixed now.